wikimedia/wikipedia-ios
# Remove dependabot warnings

**Fixes Phabricator ticket:** 
Initial work for https://phabricator.wikimedia.org/T266837

### Notes
We have been a bit stuck on dealing with these Github warnings because the version of [grunt-browserify](https://www.npmjs.com/package/grunt-browserify) hasn't updated it's dependencies in 3 years. From what I gather this package gives you the ability to break up your javascript into clean separate files, reference each other using the Node-esq `require` methods, then when you run `grunt` it combines them into 1 file into the assets folder.

What I did here is remove `grunt-browserify` from our package.json and Gruntfile and manually combined that code as they would have been combined ultimately in the assets folder. Now running the grunt command will simply copy those files from the `www` folder into the `assets` folder.

What we lose with this is the ease of working in separated files for some codemirror text detection logic and the client-side javascript we still run for the article screen (areas are listed in test steps below). We also lose some easy namespacing, so a lot of the methods and variables in these combined files are global now. I feel like generally fewer dependencies are better but giant asterisk that this stuff is definitely not in my wheelhouse. Happy to cancel this PR if anyone feels like this isn't the right path.

Potential followup is to remove this whole grunt setup entirely. We would lose the CSS compression and linting of js files.

TLDNR:

1. Updated `package.json` in the `www` folder to latest available packages.
2. Combined the 9 `codemirror-range-*.js` files in the `www/codemirror` folder into `www/codemirror-range-determination-bundle.js`. 
3. Combined the 4 files (`utilities.js`, `findInPage.js`, `elementLocation.js`, `editTextSelection.js`) in the `www/js` folder into `www/index.js`
4. Added command to include these new combined js files in the copy command section of Gruntfile (along with `about.js`, which only seemed to do a copy before anyways through browserify).
5. Removed browserify stuff from Gruntfile and `package.json`.

### Test Steps
Perform regression on article, editing and the about page. Confirm javascript commands still work which from what I could tell were:
         1. Tapping a TOC item, it scrolls properly to section.
         2. Scrolling in article across multiple sections, TOC item properly highlights
         3. Find and highlight still works
         4. Select text in the article. Tap "Edit" from the context menu. Confirm for the most part text is properly highlighted in the editing flow. (If it doesn't, confirm it does in `main` before marking it as a bug, sometimes it just fails to find the text).
         5. After highlighting text on the editing screen, tap the remove formatting (no sign) button from the keyboard menu. Confirm formatting removes as well as it did before.
         6. Confirm the About app page themes properly.
